__tests__/classes/PaymentPreprocessor.js

```diff
@@ -0,0 +1,121 @@
+    await ah.api.models.member.destroy({where: {}, force: true})
+  })
+
+  test('should add billing member in payment members when single payment is made', async () => {
+    await ah.api.models.member.create(generateMember())
+    const member2 = await ah.api.models.member.create(generateMember())
+    const payment = generatePayment({
+      billingMemberId: member2.id,
+      family: '0'
+    })
+    delete payment.members
+    const processed = await processPayment(payment)
+    expect(processed.members.length).toBe(1)
+    expect(processed.members[0]).toBe(member2.id)
+  })
+
+  test('should not add family members in payment members when single payment is made', async () => {
+    await ah.api.models.member.create(generateMember())
+    const billingMember = await ah.api.models.member.create(generateMember())
+    const family1 = await ah.api.models.member.create(generateMember())
+    const family2 = await ah.api.models.member.create(generateMember())
+    await billingMember.setFamilyMembers([family1, family2])
+
+    const payment = generatePayment({
+      billingMemberId: billingMember.id,
+      family: ''
+    })
+    delete payment.members
+
+    const processed = await processPayment(payment)
+    expect(processed.members.length).toBe(1)
+    expect(processed.members).toEqual(expect.arrayContaining([billingMember.id]))
+  })
+
+  test('should add billing member in payment members when family payment is made', async () => {
+    await ah.api.models.member.create(generateMember())
+    const billingMember = await ah.api.models.member.create(generateMember())
+    const family1 = await ah.api.models.member.create(generateMember())
+    const family2 = await ah.api.models.member.create(generateMember())
+    await billingMember.setFamilyMembers([family1, family2])
+
+    const payment = generatePayment({
+      billingMemberId: billingMember.id,
+      family: '1'
+    })
+    delete payment.members
+
+    const processed = await processPayment(payment)
+    expect(processed.members).toEqual(expect.arrayContaining([billingMember.id]))
+  })
+
+  test('should add family members in payment members when family payment is made', async () => {
+    await ah.api.models.member.create(generateMember())
+    const billingMember = await ah.api.models.member.create(generateMember())
+    const family1 = await ah.api.models.member.create(generateMember())
+    const family2 = await ah.api.models.member.create(generateMember())
+    await billingMember.setFamilyMembers([family1, family2])
+
+    const payment = generatePayment({
+      billingMemberId: billingMember.id,
+      family: '1',
+      paymentDate: '2018-05-05'
+    })
+    delete payment.members
+
+    const processed = await processPayment(payment)
+    expect(processed.members).toEqual(expect.arrayContaining([family1.id, family2.id]))
+  })
+
+  test('should add only family members who became members before the payment date', async () => {
```

Comment:
this could be split in three:

1. should add family members who joined before payment date
2. should not add family members who joined after payment date
3. should add family members who joined **on** payment date